src/main/java/org/broadinstitute/hellbender/tools/funcotator/simpletsvoutput/SimpleTsvOutputRenderer.java

```diff
@@ -0,0 +1,366 @@
+        Utils.nonNull(unaccountedForOverrideAnnotations);
+        Utils.nonNull(excludedOutputFields);
+        Utils.nonNull(columnNameToAliasesMap);
+        Utils.nonNull(toolVersion);
+
+        this.excludedOutputFields = excludedOutputFields;
+        this.outputFilePath = outputFilePath;
+        this.isWriterInitialized = false;
+        this.aliasProvider = new AliasProvider(columnNameToAliasesMap);
+
+        this.unaccountedForDefaultAnnotations = unaccountedForDefaultAnnotations;
+        this.unaccountedForOverrideAnnotations = unaccountedForOverrideAnnotations;
+
+        this.isWriteFuncotationFieldsNotInConfig = isWriteFuncotationFieldsNotInConfig;
+    }
+
+    // This method uses the fact that the keys of the input are ordered.
+    private void initializeWriter(final List<String> columnNames) {
+        Utils.validateArg(columnNames.size() > 0, "TSV output renderer has been configured to produce a blank file.  This is usually a user error.  Please check excluded columns.");
+        final TableColumnCollection columns = new TableColumnCollection(columnNames);
+        try {
+            writer = TableUtils.writer(outputFilePath, columns, (map, dataLine) -> {
+                map.keySet().forEach(k -> dataLine.set(k, map.get(k)));
+            });
+        } catch (final IOException ioe) {
+            throw new GATKException("Could not open the simple TSV writer.", ioe);
+        }
+        isWriterInitialized = true;
+    }
+
+    @Override
+    public void close() {
+        try {
+            // Initialize the writer (so that we can write column names and do not just produce a blank file when
+            //  write was never called).
+            if (!isWriterInitialized) {
+                final List<String> columnNames = estimateOutputFieldsWithoutData(excludedOutputFields, unaccountedForDefaultAnnotations,unaccountedForOverrideAnnotations);
+                initializeWriter(columnNames);
+            }
+            if (writer != null) {
+                writer.close();
+            }
+        } catch (final IOException ioe) {
+            throw new GATKException("Could not close the simple TSV output writing.", ioe);
+        }
+    }
+
+    @Override
+    public void write(final VariantContext variant, final FuncotationMap txToFuncotationMap) {
+        if (txToFuncotationMap.getTranscriptList().size() > 1) {
+            logger.warn("More than one transcript found.  This should be able to render (grouped by transcript), but you may need to do further processing.  No user action needed.");
+        }
+
+        // Ensure that all transcript-allele combinations have the same fields inside the matching funcotations.
+        if (!txToFuncotationMap.doAllTxAlleleCombinationsHaveTheSameFields()) {
+            throw new GATKException.ShouldNeverReachHereException("The funcotation map cannot be written by this simple output renderer.  This is almost certainly an issue for the GATK development team.");
```

Comment:
Can you make the error message include the fact that the fields differ?

Reply:
Done